src/components/user-inputs.js

```diff
@@ -9,8 +9,24 @@ function registerOn (browserWindow, { viewState }) {
    * Called whenever the mouse is pressed
    */
   function mousePressed () {
+    if (browserWindow.mouseX < 0 || browserWindow.mouseX > viewState.canvasWidth) {
+      return
+    } if (browserWindow.mouseY < 0 || browserWindow.mouseY > viewState.canvasHeight) {
+      return
+    }
```

Comment:
Suggest a bit of a refactor, thoughts?
```js
  function isMouseOutOfBounds({ mouseX, mouseY }) {
    return (mouseX < 0 || mouseX > viewState.canvasWidth || mouseY < 0 || mouseY > viewState.canvasHeight)
  }

  /**
   * P5 MousePressed
   *
   * Called whenever the mouse is pressed
   */
  function mousePressed () {
    if (isMouseOutOfBounds(browserWindow)) {
      return
    }

    viewState.clickX = browserWindow.mouseX
    viewState.clickY = browserWindow.mouseY
    viewState.dragActive = true
  }
```

Reply:
I like this refactor, yes. Makes the change more planned and readable!

Reply:
Are you ok making that change on your branch?